staging/src/k8s.io/apiserver/pkg/authentication/serviceaccount/util.go

```diff
@@ -42,6 +42,15 @@ const (
 	// PodUIDKey is the key used in a user's "extra" to specify the pod UID of
 	// the authenticating request.
 	PodUIDKey = "authentication.kubernetes.io/pod-uid"
+	// JTIKey is the key used in a user's "extra" (to specify the JTI of the authenticating request)
+	// and audit event annotations (to specify the JTI of the newly created SA token).
+	JTIKey = "authentication.kubernetes.io/jti"
```

Comment:
@liggitt do we want this to be `token-identifier` or `credential-identifier`?

Reply:
this is continuing [this thread](https://github.com/kubernetes/enhancements/pull/4141#discussion_r1335950901) from the KEP

cc @enj @sftim 

The most likely expansion direction in the future would be service account client certificates if we ever add those.

For the audit log, I could see having more info or splitting authn info by type (since audit data is structured). cc @tallclair

For user extra info, we should:
1. be concise
2. give enough information to correlate to the audit logs for requests when the credential is used
3. give enough information to correlate to the audit log for the request where the credential was issued (if kubernetes issued the credential) – e.g. the tokenrequest or the CertificateSigningRequest update with the issued certificate

For tokens, the minimal information to correlate is the jti claim

For certs, minimal information to correlate could either be a hash or issuer + serial (issuer + serial is what's included in certificate revocation lists, fwiw).

Ideally, 2 and 3 would be generic (I could select relevant audit events based on extra userInfo without splitting my lookup based on the credential type).

For userInfo extra data, I would lean towards making this generic (like `authentication.kubernetes.io/credential-id`) and setting the value when populated from a JTI like `JTI=$jti` (that fits nicely with RDF formatting if we do $issuer+$serial in the future for cert auth)

Reply:
> I think the question of including further info beyond that, may need some more thought.

Agreed, and it doesn't need to be decided here.

> If the goals are to make it easy/possible to take any identity document and cross reference that document with entries in the audit log, would we extend the SubjectAccessReview API to return the computed credential-id for a presented document?

I don't understand that... SubjectAccessReview is answering authz questions, not authn questions.

Reply:
so... switch this to `CredentialIDKey = "authentication.kubernetes.io/credential-id"` and make a helper function to turn a JTI into a credential id value (maybe just prefix with `JTI=`)?